src/payment/data/sagas.js

```diff
@@ -88,7 +88,7 @@ export function* handleDiscountCheck() {
         discount.jwt,
       );
       yield put(basketDataReceived(result));
-      yield call(handleMessages, result.messages, true);
+      yield call(handleMessages, result.messages, false);
```

Comment:
I tweaked this for efficiency - we were calling clearMessages() more often than we needed to.